workspace.spec

```diff
@@ -572,6 +572,497 @@ module Workspace {
+
+		In the following documentation, 'Resource' is used to refer to the workspace object
+		that the CM pertains to.
+
+		The 'resource_type' field should be filled using values from the DataCite
+		resourceTypeGeneral field:
+
+		https://support.datacite.org/docs/datacite-metadata-schema-v44-mandatory-properties#10a-resourcetypegeneral
+
+		Currently the KBase workspace only supports credit metadata for objects of type
+		'dataset'; anything else will return an error.
+
+		The license may be supplied either as an URL pointing to licensing information for
+		the resource, or using a license name. See https://choosealicense.com/appendix/ for
+		a list of common open source licenses.
+
+		Required fields are:
+		- identifier
+		- resource_type
+		- versioning information: if the resource does not have an explicit version number,
+		one or more dates should be supplied: ideally the date of resource publication and
+		the last update (if applicable).
+		- contributors (one or more required)
+		- titles (one or more required)
+
+		comments - freeform text providing extra information about this credit metadata.
+			Examples:
+				- Credit metadata generated automatically from DOI:10.13039/100000015
+
+		identifier - persistent unique identifier for the resource (i.e. the source data
+			for this workspace object).
+			Should be in the format <database name>:<identifier within database>
+			Examples:
+				- RefSeq:GCF_004214875.1
+				- GenBank:CP035949.1
+				- img.taxon:648028003
+
+		license (optional) - usage license for the resource. May be a text string or an
+			URL. Abbreviations should be spelled out where possible (e.g. 'Creative
+			Commons 4.0' instead of 'CC-BY-4.0').
+			Examples:
+				- Creative Commons 4.0
+				- MIT
+				- https://jgi.doe.gov/user-programs/pmo-overview/policies/
+
+		resource_type - the broad type of the source data for this workspace object.
+			Defaults to 'dataset' (the only valid value currently) if no value is
+			provided.
+			Valid 'resource_type' values:
+				- dataset
+
+		version (optional if dates are provided) - the version of the resource. This must
+			be an absolute version, not a relative version like 'latest'.
+			Examples:
+				- 5
+				- 1.2.1
+				- 20220405
+
+		contributors - a list of people and/or organizations who contributed to the
+			resource.
+
+		dates (optional if version is provided) - a list of relevant lifecycle events for
+			the resource.
+
+		funding (optional) - funding sources for the resource.
+
+		related_identifiers (optional) - other resolvable persistent unique IDs related to
+			the resource.
+
+		titles - one or more titles for the resource.
+
+	*/
+	typedef structure {
+		string comments;
+		string identifier;
+		string license;
+		string resource_type;
+		string version;
+		list<Contributor> contributors;
+		list<EventDate> dates;
+		list<FundingReference> funding;
+		list<PermanentID> related_identifiers;
+		list<Title> titles;
+	} CreditMetadata;
```

Comment:
We should decide on and document a maximum size for the CM when serialized

Reply:
I was thinking we should do that in this PR - did you want to delay?

Reply:
yes -- was going to put in the documentation when the size check is implemented

Reply:
Ok, as long as it winds up in the spec at some point

:+1: